### PR TITLE
New Strategies: Mem2, Pun1, Collective Strategy

### PR DESCRIPTION
**Pull request by marcharper**
All from the literature.

### CODE REVIEW

File: axelrod/strategies/finite_state_machines.py [hunk 1]
```diff
@@ -148,28 +153,56 @@ class Predator(FSMPlayer):
...
+            (7, D, 7, D),
+            (8, C, 8, D),
+            (8, D, 6, D)
+        )
 
         FSMPlayer.__init__(self, transitions, initial_state=1, initial_action=C)
 
 
+class Pun1(FSMPlayer):
+    """FSM player described in [Ashlock2006]."""
```
Comment (drvinceknight) [inline axelrod/strategies/finite_state_machines.py]: Could you write this docstrings to match up with the convention (not all of them follow it yet, slow conversion :)). In particular for this it should have (appended to what is already there).

```
Names

    - Pun1 [Aslhlock2006]_
```

File: axelrod/strategies/memorytwo.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+class MEM2(Player):
+    """A memory-two player that switches between TFT, TFTT, and ALLD.
+
+    Note that the reference claims that this is a memory two strategy but in
+    fact it is infinite memory. This is because the player plays as ALLD if
+    ALLD has ever been selected twice, which can only be known if the entire
+    history of play is accessible.
+
+    Names
+    -----
```
Comment (drvinceknight) [inline axelrod/strategies/memorytwo.py]: No `-----`, just:

```
Names:

    - ...
```

File: axelrod/strategies/prober.py [hunk 3]
```diff
@@ -5,6 +5,42 @@
...
 
 
+class CollectiveStrategy(Player):
+    """Defined in [Li2009]. 'It always cooperates in the first move and defects
+    in the second move. If the opponent also cooperates in the first move and
+    defects in the second move, CS will cooperate until the opponent defects.
+    Otherwise, CS will always defect.'
+
+    Names:
+        Collective Strategy [Li2009]
```
Comment (drvinceknight) [inline axelrod/strategies/prober.py]:
```
    - Collective Strategy
````

Comment (marcharper) [inline axelrod/strategies/prober.py]: Done

**Pull request merged**